drivers/i3c/i3c_cdns.c

```diff
@@ -923,6 +923,12 @@ static void cdns_i3c_start_transfer(const struct device *dev)
 	struct cdns_i3c_data *data = dev->data;
 	const struct cdns_i3c_config *config = dev->config;
 	struct cdns_i3c_xfer *xfer = &data->xfer;
+	uint32_t retry = 10000;
+
+	/* poll idle register, ensuring that the core already completed it's last cmd */
+	while (!(sys_read32(config->base + MST_STATUS0) & MST_STATUS0_IDLE) && (retry > 0)) {
+		retry--;
+	}
```

Comment:
Do you want to return error if this runs out of retries?

Reply:
yeah... I probably should return -EAGAIN